spacy/util.py

```diff
@@ -898,11 +898,17 @@ def escape_html(text):
 def use_gpu(gpu_id):
     return require_gpu(gpu_id)
 
+def gpu_is_available():
+    try:
+        cupy.cuda.runtime.getDeviceCount()
+        return True
+    except cupy.cuda.runtime.CUDARuntimeError:
+        return False
 
 def fix_random_seed(seed=0):
     random.seed(seed)
     numpy.random.seed(seed)
-    if cupy is not None:
+    if cupy is not None and gpu_is_available():
```

Comment:
I added this because I had a GPU machine with cupy installed but was trying to use CPU training.